lib/lfrfid/protocols/protocol_securakey.c

```diff
@@ -0,0 +1,301 @@
+
+    // 32-bit format (1-bit EP, 14-bit facility number, 16-bit card number, 1-bit OP)
+    // preamble     ??bitlen   reserved  EPfffffff   fffffffc   cccccccc   cccccccOP  CS?        CS2?
+    // 0111111111 0 01100000 0 00000000 0 10000100 0 11001010 0 01011011 0 01010110 0 00010110 0 11100000 0 00000000 0 0000
+
+    // get bit length (26-bit or 32-bit)
+    // left two 0 paddings in the beginning for easier parsing (00011010 = 011010)
+    bit_lib_copy_bits(protocol->data, 2, 6, protocol->encoded_data, 13); 
+    
+    // get facility number (f)
+    if(bit_lib_get_bits(protocol->data, 0, 8) == 26) {
+        FURI_LOG_D(TAG,"26-bit Securakey detected");
+        protocol->bit_format = 26;
+        bit_lib_copy_bits(protocol->data, 16, 1, protocol->encoded_data, 36);
+            // have to skip one spacer
+        bit_lib_copy_bits(protocol->data, 17, 7, protocol->encoded_data, 38);
+    } else if(bit_lib_get_bits(protocol->data, 0, 8) == 32) {
+        FURI_LOG_D(TAG,"32-bit Securakey detected");
+        protocol->bit_format = 32;
+        // same two 0 paddings here, otherwise should be bit_lib_copy_bits(protocol->data, 8, 7, protocol->encoded_data, 30);
+        bit_lib_copy_bits(protocol->data, 10, 7, protocol->encoded_data, 30);
+            // have to skip one spacer
+        bit_lib_copy_bits(protocol->data, 17, 7, protocol->encoded_data, 38);
+    }
+ 
+    // get card number (c)
+    bit_lib_copy_bits(protocol->data, 24, 1, protocol->encoded_data, 45); 
+        // same skips here
+    bit_lib_copy_bits(protocol->data, 25, 8, protocol->encoded_data, 47);
+    bit_lib_copy_bits(protocol->data, 33, 7, protocol->encoded_data, 56);
+
+    // unsure about CS yet, might as well just save it
+        // CS1
+    bit_lib_copy_bits(protocol->data, 40, 8, protocol->encoded_data, 65);
+        // CS2
+    bit_lib_copy_bits(protocol->data, 48, 8, protocol->encoded_data, 74);
+    //FURI_LOG_D(TAG, "decoded data = %s", protocol_securakey_get_data_debug(protocol));
+
+    // (decoded) data looks like this (pp are zero paddings):
+    // 26-bit format (1-bit EP,  8-bit facility number, 16-bit card number, 1-bit OP)
+    // ppbitlen pppppppp ffffffff cccccccc cccccccc CS1      CS2
+    // 00011010 00000000 00011011 00011111 00110001 00001111 01100000
+
+    // 32-bit format (1-bit EP, 14-bit facility number, 16-bit card number, 1-bit OP)
+    // ppbitlen ppffffff ffffffff cccccccc cccccccc CS1      CS2?
+    // 00100000 00000010 01100101 00101101 10101011 00010110 11100000
+};
+
+void protocol_securakey_decoder_start(ProtocolSecurakey* protocol) {
+    memset(protocol->encoded_data, 0, SECURAKEY_ENCODED_FULL_SIZE_BYTE);
+    manchester_advance(
+        protocol->decoder_manchester_state,
+        ManchesterEventReset,
+        &protocol->decoder_manchester_state,
+        NULL);
+};
+
+bool protocol_securakey_decoder_feed(ProtocolSecurakey* protocol, bool level, uint32_t duration) {
```

Comment:
Please use `./fbt format` to format sources

Reply:
Done. Thanks for introducing that to me! I didn't know it can do this.